src/scores.jl

```diff
@@ -1,83 +1,105 @@
 
+A DistrictAggregate score is a simple sum of a particular property
+over all nodes in a given district.
+"""
 struct DistrictAggregate <: AbstractScore
-    """ A DistrictAggregate score is a simple sum of a particular property
-        over all nodes in a given district.
-    """
     name::String
     key::String
 end
 
+"""
+    DistrictScore(name::Union{String, Missing}
+                  score_fn::Function)
 
+A `DistrictScore` takes a user-supplied function that returns some
+quantity of interest given the nodes in a given district. The signature
+of `score_fn` should be as follows:
+    `score_fn(graph::BaseGraph, district_nodes::BitSet, district::int)`
+"""
 struct DistrictScore <: AbstractScore
-    """ A DistrictScore takes a user-supplied function that returns some
-        quantity of interest given the nodes in a given district. The signature
-        of `score_fn` should be as follows:
-            score_fn(graph::BaseGraph, district_nodes::BitSet, district::int)
-    """
     name::Union{String, Missing}
     score_fn::Function
     DistrictScore(score_fn::Function) = new(missing, score_fn)
     DistrictScore(name::String, score_fn::Function) = new(name, score_fn)
 end
 
+"""
+    PlanScore(name::Union{String, Missing},
+              score_fn::Function)
 
+A `PlanScore` takes a user-supplied function that returns some quantity of
+interest given a `BaseGraph` and corresponding `Partition` object.
+
+The signature of `score_fn` should be as follows:
+    `score_fn(graph::BaseGraph, partition::Partition)`
+"""
 struct PlanScore <: AbstractScore
-    """ A PlanScore takes a user-supplied function that returns some
-        quantity of interest given a Graph and corresponding Partition object.
-        The signature of `score_fn` should be as follows:
-            score_fn(graph::BaseGraph, partition::Partition)
-    """
     name::Union{String, Missing}
     score_fn::Function
     PlanScore(score_fn::Function) = new(missing, score_fn)
     PlanScore(name::String, score_fn::Function) = new(name, score_fn)
 end
 
+"""
+    CompositeScore(name::String,
+                   scores::Array{S,1}) where {S<:AbstractScore} # should be other AbstractScores
 
+A `CompositeScore` is just a group of scores that are run in sequence.
+`CompositeScore`s are especially useful when the score functions depend
+upon/modify some shared state.
+"""
 struct CompositeScore <: AbstractScore
-    """ A CompositeScore is just a group of scores that are run in sequence.
-        CompositeScores are especially useful when the score functions depend
-        upon/modify some shared state.
-    """
     name::String
     scores::Array{S,1} where {S<:AbstractScore} # should be other AbstractScores
 end
 
 
+"""
+    ChainScoreData(scores::Array{S,1} where {S<:AbstractScore} # scores that were measured on a particular chain
+                   step_values::Array{Dict{String, Any}}) # array of Dicts which map {score name: score value}
```

Comment:
Is this the right format for this docstring? just checking

Reply:
Great question... we actually do not expose this particular docstring in the API so I don't know - you can get `ChainScoreData` from `get_scores()` and `get_scores_at_step()` but we don't expect the user to ever instantiate it themself (should we? i think it is fine as is for now.)